pkg/gadgets/profile/tcprtt/tracer/tracer_test.go

```diff
@@ -505,7 +589,352 @@ func TestRunWithResult(t *testing.T) {
+			for i, l := range result.Histograms {
+				t.Logf("Result [%d/%d]:", i+1, lHis)
+				t.Logf("AddrType %s - Addr %s - Avg %f", l.AddressType, l.Address, l.Average)
+				t.Logf("Histogram: %s", l.Histogram)
+			}
+
+			// Validate the number of histograms
+			if test.expectedResult.exactHistograms != 0 {
+				require.Equal(t, test.expectedResult.exactHistograms, lHis,
+					"invalid number of histograms")
+			} else {
+				// When we don't filter by address, we can't know how many
+				// histograms will be generated. This is because the number of
+				// histograms depends on the number of connections that are
+				// established in the host during the test. Therefore, we can
+				// only check that the number of histograms is greater or equal
+				// than the minimum number of histograms we expect.
+				require.GreaterOrEqual(t, lHis, test.expectedResult.minHistograms,
+					"wrong number of histograms")
+			}
+
+			lAddrFound := false
+			rAddrFound := false
+			for _, h := range result.Histograms {
+				// Validate unit
+				var expectedUnit histogram.Unit
+				if test.expectedResult.useMilliseconds {
+					expectedUnit = histogram.UnitMilliseconds
+				} else {
+					expectedUnit = histogram.UnitMicroseconds
+				}
+				require.Equal(t, expectedUnit, h.Unit, "wrong unit for histogram")
+
+				// Validate histogram contains at least one interval
+				require.Greater(t, len(h.Intervals), int(0),
+					"expecting at least one interval")
+
+				// Validate histogram computed the average value
+				// TODO: Verify this also when using milliseconds once the BPF
+				// program will be able to report the total latencies between 0
+				// and 1. Otherwise, the test will fail because the average
+				// latency will always be 0 milliseconds, so there is no way to
+				// verify that it was computed correctly.
+				if !test.expectedResult.useMilliseconds {
+					require.Greater(t, h.Average, float64(0))
+				}
+
+				// Validate histogram contains the expected address information
+				if test.expectedResult.byLocalAddr != "" {
+					require.Equal(t, types.AddressTypeLocal, h.AddressType)
+
+					if h.Address == test.expectedResult.byLocalAddr {
+						lAddrFound = true
+					}
+				} else if test.expectedResult.byRemoteAddr != "" {
+					require.Equal(t, types.AddressTypeRemote, h.AddressType)
+
+					if h.Address == test.expectedResult.byRemoteAddr {
+						rAddrFound = true
+					}
+				} else {
+					require.Equal(t, types.AddressTypeAll, h.AddressType)
+					require.Equal(t, types.WildcardAddress, h.Address)
+				}
+			}
+			if test.expectedResult.byLocalAddr != "" {
+				require.True(t, lAddrFound,
+					"expected to find histogram with local address %s", test.expectedResult.byLocalAddr)
+			} else if test.expectedResult.byRemoteAddr != "" {
+				require.True(t, rAddrFound,
+					"expected to find histogram with remote address %s", test.expectedResult.byRemoteAddr)
+			}
+		})
+	}
+}
+
+func TestRunWithResultV6(t *testing.T) {
```

Comment:
I made this work locally using the loopback address `::1`. Did you try it? The problem with using the loopback address is that IPv6 allocates only one single loopback address, so we would need to use the same for the client and server. 

In addition, I found that some tests are reporting data for `127.0.0.1` too. I didn't investigate the reason, but here is the output:
<details>
  <summary>Tests output</summary>
  
  ```bash
$ sudo go test -run TestRunWithResultV6 ./pkg/gadgets/profile/tcprtt/...
--- FAIL: TestRunWithResultV6 (0.00s)
    --- FAIL: TestRunWithResultV6/mix_by_local_and_filter_by_laddr_v6 (4.07s)
        tracer_test.go:970: Result [1/2]:
        tracer_test.go:971: AddrType Local - Addr :: - Avg 10.731959
        tracer_test.go:972: Histogram:         µs               : count    distribution
                     0 -> 1          : 0        |                                        |
                     2 -> 3          : 0        |                                        |
                     4 -> 7          : 34       |**************************              |
                     8 -> 15         : 51       |****************************************|
                    16 -> 31         : 9        |*******                                 |
                    32 -> 63         : 3        |**                                      |
        tracer_test.go:970: Result [2/2]:
        tracer_test.go:971: AddrType Local - Addr 127.0.0.1 - Avg 91.954545
        tracer_test.go:972: Histogram:         µs               : count    distribution
                     0 -> 1          : 0        |                                        |
                     2 -> 3          : 0        |                                        |
                     4 -> 7          : 0        |                                        |
                     8 -> 15         : 0        |                                        |
                    16 -> 31         : 0        |                                        |
                    32 -> 63         : 8        |************                            |
                    64 -> 127        : 25       |****************************************|
                   128 -> 255        : 11       |*****************                       |
        tracer_test.go:977:
                Error Trace:    /inspektor-gadget/pkg/gadgets/profile/tcprtt/tracer/tracer_test.go:977
                Error:          Not equal:
                                expected: 1
                                actual  : 2
                Test:           TestRunWithResultV6/mix_by_local_and_filter_by_laddr_v6
                Messages:       invalid number of histograms
    --- FAIL: TestRunWithResultV6/mix_by_remote_and_filter_by_laddr_v6 (4.09s)
        tracer_test.go:970: Result [1/2]:
        tracer_test.go:971: AddrType Remote - Addr ::1 - Avg 72.680851
        tracer_test.go:972: Histogram:         µs               : count    distribution
                     0 -> 1          : 0        |                                        |
                     2 -> 3          : 0        |                                        |
                     4 -> 7          : 0        |                                        |
                     8 -> 15         : 5        |*****                                   |
                    16 -> 31         : 28       |****************************            |
                    32 -> 63         : 39       |****************************************|
                    64 -> 127        : 11       |***********                             |
                   128 -> 255        : 5        |*****                                   |
                   256 -> 511        : 5        |*****                                   |
                   512 -> 1023       : 1        |*                                       |
        tracer_test.go:970: Result [2/2]:
        tracer_test.go:971: AddrType Remote - Addr 10.0.2.2 - Avg 370.000000
        tracer_test.go:972: Histogram:         µs               : count    distribution
                     0 -> 1          : 0        |                                        |
                     2 -> 3          : 0        |                                        |
                     4 -> 7          : 0        |                                        |
                     8 -> 15         : 0        |                                        |
                    16 -> 31         : 0        |                                        |
                    32 -> 63         : 0        |                                        |
                    64 -> 127        : 0        |                                        |
                   128 -> 255        : 0        |                                        |
                   256 -> 511        : 2        |****************************************|
        tracer_test.go:977:
                Error Trace:    /inspektor-gadget/pkg/gadgets/profile/tcprtt/tracer/tracer_test.go:977
                Error:          Not equal:
                                expected: 1
                                actual  : 2
                Test:           TestRunWithResultV6/mix_by_remote_and_filter_by_laddr_v6
                Messages:       invalid number of histograms
    --- FAIL: TestRunWithResultV6/mix_by_remote_and_filter_by_raddr_v6 (4.09s)
        tracer_test.go:970: Result [1/2]:
        tracer_test.go:971: AddrType Remote - Addr 127.0.0.1 - Avg 68.035714
        tracer_test.go:972: Histogram:         µs               : count    distribution
                     0 -> 1          : 0        |                                        |
                     2 -> 3          : 0        |                                        |
                     4 -> 7          : 0        |                                        |
                     8 -> 15         : 2        |****                                    |
                    16 -> 31         : 0        |                                        |
                    32 -> 63         : 9        |*********************                   |
                    64 -> 127        : 17       |****************************************|
        tracer_test.go:970: Result [2/2]:
        tracer_test.go:971: AddrType Remote - Addr ::1 - Avg 60.263158
        tracer_test.go:972: Histogram:         µs               : count    distribution
                     0 -> 1          : 0        |                                        |
                     2 -> 3          : 0        |                                        |
                     4 -> 7          : 0        |                                        |
                     8 -> 15         : 6        |****                                    |
                    16 -> 31         : 6        |****                                    |
                    32 -> 63         : 56       |****************************************|
                    64 -> 127        : 24       |*****************                       |
                   128 -> 255        : 3        |**                                      |
        tracer_test.go:977:
                Error Trace:    /inspektor-gadget/pkg/gadgets/profile/tcprtt/tracer/tracer_test.go:977
                Error:          Not equal:
                                expected: 1
                                actual  : 2
                Test:           TestRunWithResultV6/mix_by_remote_and_filter_by_raddr_v6
                Messages:       invalid number of histograms
    --- FAIL: TestRunWithResultV6/by_local_addr (4.05s)
        tracer_test.go:970: Result [1/2]:
        tracer_test.go:971: AddrType Local - Addr :: - Avg 62.916667
        tracer_test.go:972: Histogram:         µs               : count    distribution
                     0 -> 1          : 0        |                                        |
                     2 -> 3          : 0        |                                        |
                     4 -> 7          : 0        |                                        |
                     8 -> 15         : 0        |                                        |
                    16 -> 31         : 4        |***                                     |
                    32 -> 63         : 53       |****************************************|
                    64 -> 127        : 33       |************************                |
                   128 -> 255        : 6        |****                                    |
        tracer_test.go:970: Result [2/2]:
        tracer_test.go:971: AddrType Local - Addr 127.0.0.1 - Avg 78.956522
        tracer_test.go:972: Histogram:         µs               : count    distribution
                     0 -> 1          : 0        |                                        |
                     2 -> 3          : 0        |                                        |
                     4 -> 7          : 0        |                                        |
                     8 -> 15         : 0        |                                        |
                    16 -> 31         : 0        |                                        |
                    32 -> 63         : 11       |************************************    |
                    64 -> 127        : 12       |****************************************|
        tracer_test.go:1035:
                Error Trace:    /inspektor-gadget/pkg/gadgets/profile/tcprtt/tracer/tracer_test.go:1035
                Error:          Should be true
                Test:           TestRunWithResultV6/by_local_addr
                Messages:       expected to find histogram with local address ::1
    --- FAIL: TestRunWithResultV6/mix_by_local_and_filter_by_raddr_v6 (4.05s)
        tracer_test.go:970: Result [1/2]:
        tracer_test.go:971: AddrType Local - Addr :: - Avg 62.916667
        tracer_test.go:972: Histogram:         µs               : count    distribution
                     0 -> 1          : 0        |                                        |
                     2 -> 3          : 0        |                                        |
                     4 -> 7          : 0        |                                        |
                     8 -> 15         : 0        |                                        |
                    16 -> 31         : 4        |***                                     |
                    32 -> 63         : 53       |****************************************|
                    64 -> 127        : 33       |************************                |
                   128 -> 255        : 6        |****                                    |
        tracer_test.go:970: Result [2/2]:
        tracer_test.go:971: AddrType Local - Addr 127.0.0.1 - Avg 78.956522
        tracer_test.go:972: Histogram:         µs               : count    distribution
                     0 -> 1          : 0        |                                        |
                     2 -> 3          : 0        |                                        |
                     4 -> 7          : 0        |                                        |
                     8 -> 15         : 0        |                                        |
                    16 -> 31         : 0        |                                        |
                    32 -> 63         : 11       |************************************    |
                    64 -> 127        : 12       |****************************************|
        tracer_test.go:977:
                Error Trace:    /inspektor-gadget/pkg/gadgets/profile/tcprtt/tracer/tracer_test.go:977
                Error:          Not equal:
                                expected: 1
                                actual  : 2
                Test:           TestRunWithResultV6/mix_by_local_and_filter_by_raddr_v6
                Messages:       invalid number of histograms
FAIL
FAIL    github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/profile/tcprtt/tracer  17.308s
?       github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/profile/tcprtt/types   [no test files]
FAIL
  ```
</details>

Reply:
Sorry, I forgot to share the changes I did: https://github.com/inspektor-gadget/inspektor-gadget/commit/b83ee93ee8146a64f87bd2e61cb9d207c93dac71